<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1">
<meta name="author" content="VSN International Ltd">
<meta name="copyright" content="Copyright&nbsp;&copy; 2010-2015 VSN International Ltd.">
<meta name="language" content="en">
<title>ASReml v4.1: Release Notes</title>
<style>
body
        {
        background-color: #006699;
        color: black;
        font-family: Helvetica, Arial, sans-serif ;
        font-size: 10pt;
        font-weight: normal;
        font-style: normal;
        }

h2
	{
	color: #000066;
	}
h3
	{
	color: #000066;
	}
</style>
</head>

<body>
<table align="center" width="760" cellspacing="0" cellpadding="0" bgcolor="#ffffff">
<tr><td>
<table cellpadding="0" cellspacing="0" width="100%"><tr style="height:20px"><td/></tr><tr><td style="width:20px"></td><td valign="top">
</td><td width="10"></td><td bgcolor="#b0b0c3" width="1"></td><td width="10"></td><td valign="top" width="620px">
<h2>ASReml v4.1: Release Notes</h2>


<h3>ASReml 4 Version History</h3>
<ul>
<li>ASReml 4.1 is the current released version</li>
<li>The documentation folder contains the UpdateR4.pdf file that identifies what's new in Release 4.1</li>
<li>In addition there are two ASReml 4.1 User Guide document files available :- UserGuideStructural.pdf and UserGuideFunctional.pdf respectively</li>
</ul>
<h4>v4.1my (31st March 2019)</h4>
<ul>
<li>Minor efficiency improvements.</li>
</ul>
<h4>v4.1mx (30th September 2018)</h4>
<ul>
<li>!TAU is allowed as an alias for !Beta</li>
<li>A bug in parsing xfa1(leg(bbb,1)).Subject has been fixed.</li>
<li>A bug in UPCASE (overflow linelength) has been fixed.</li>
<li>A bug in pol/leg functions with missing X variables has been fixed.</li>
<li>A notice has been added if TSV changes constraints.</li>
<li>A bug when there are tab separated tokens in .pin file has been fixed.</li>
<li>A bug in processing at(year,2).jlast when !MVREMOVE has been fixed.</li>
<li>The order for reordering Cinverse has been changed.</li>
</ul>
<h4>v4.1mw (31st May 2018)</h4>
<ul>
<li>The use of # in is now allowed in pedigree entry names.</li>
<li>Improved .tsv error messages.</li>
<li>A problem in parsing !SUBSET has been fixed.</li>
<li>A problem when grm(Mcln) has levels coded 0 has been fixed.</li>
<li>A problem when GIV  has many more levels than base !I factor has been fixed.</li>
<li>A problem that giv referenced the wrong variable when factor defined after a GROUP factor has been fixed.</li>
<li>Fix to failure to honour second trait weights when estimating missing values.</li>
<li>A problem causing ASReml to crash while reporting the variance components when using maximum workspace has been fixed.</li>
<li>New unofficial qualifiers (for investigation) !Beta and !ARlimit.</li>
</ul>
<h4>v4.1mv (28th November 2017)</h4>
<ul>
<li>The vec() model function, which applies a set of variables (!G g) in parallel to a set of g traits, was not working when the variables were factors (!G g l) coding for l class; this has now been fixed.</li>
<li>The number of lines that report singularities in a G matrix has been reduced, reporting them as groups rather than each singularity separately.</li>
<li>A bug has been fixed, which meant ASReml failed to read a GRM matrix if it was stored full.</li>
<li>A labelling problem has been fixed, to label components properly when out of order.</li>
<li>Some warning notices have been modified to make their meaning clearer.</li>
<li>A bug, which allowed a correlation to update to zero, has been fixed; a correlation parameter of zero is not allowed.</li>
<li>Contents of the .sco file in VCM cases have been modified to report SCORE and constrained GAMMA from the final iteration.</li>
<li>Processing of 3-way models, involving reduced rank factor analytic (F) and a GRM matrix (G), has been revised. The processing rules require that such models are specified with these terms respectively first and last; i.e. F.I.G or G.I.F.  Messages reporting failure of this requirement have been updated.</li>
<li>G.A.F models can now be fitted. An updated error message is given, if a VCM matrix is not positive definite.</li>
<li>A bug has been fixed in utilizing ASSIGN strings when one string name is a shortened form of another (e.g.  using  t  and  tot).</li>
<li>There has been a general revision in the processing of VCM constraints on variance parameters, to improve handling of constraints applied to different types of parameters.</li>
<li>An integer overflow bug, that occured in calculating a Deviance table when a large amount of memory was utilized, has been fixed.</li>
<li>Adding of 'hidden' model terms, when !WRITEDESIGN is specified, is now avoided, since prediction is not performed in such jobs.</li>
</ul>
<h4>v4.1mr (31st May 2017)</h4>
<ul>
The main change in this release is to the equation ordering routine. The new routine has resulted in a 20% reduction in execution time in some jobs. You can can compare times with your own jobs by using the !EQORDER 5 qualifier to select the previous routine. Please advise support@vsni.co.uk if you find cases where the new order is significantly worse than the previous order.
<li>The notice associated with use of the !GOFFSET (pedigree groups) qualifier has changed. If this qualifier is used, it is now recommended you try several values for the parameter value (say 0.1, 1 and 10), and choose the value which gives the highest likelihood. Use of a small value when genetic group effects are small can bias the genetic variance down.</li>
<li> Previously the syntax sat(ENV).units failed when ENV had more than 999 levels. Now the limit is 9999 levels; but ASReml can not fit that many variance components.</li>
<li>ASReml now detects when a model term is actually larger than anticipated.</li>
<li>Previously ide(Sire.CG) crashed ASReml. ide() was not intended to be used this way as the number of levels may not be set correctly; the term should be specified as ide(Sire).CG
when a NRM is associated with Sire.</li>
<li>Several bugs have been fixed in relation to the !VCM qualifier. This procedure can link parameters of different types which can confuse the parameter update rules. The VCM matrix is printed to the .asl file in !DEBUG mode to facilitate checking.</li>
<li>Other bug fixes relate to spelling, mishandling of missing values in some situations, parsing !DO !ENDDO transformations and some graphical output when processing multiple parts.</li>
</ul>
<h4>v4.1mn (10th January 2017)</h4>
<ul>
This release fixes the bugs listed below.
<li>bug in !RANGE qualifier so it works with data -10 : 10.</li>
<li>Fix inconsistency in AIVAIG arg list affecting XLINK.</li>
<li>Change SLIST(100), CLIST(100) to allocatable.</li>
<li>Bug labelling !A factor when more than 90000 !A levels.</li>
<li>Bug in Stratum Var when variances large.</li>
<li>Default plot format changed to EPS.</li>
<li>Comment if var(Y) extreme.</li>
<li>Allow xfa x ar1 x ar1.</li>
<li>Bug setting up residual sat(Site).id(units).diag(2).</li>
<li>Failed to honour !S2==1 in multisection multivariate STRUCTURAL w/o Start values</li>
<li>Failed to reset REREAD in MS MV STRUCTURAL w/o SV.</li>
</ul>
<h4>v4.1mj (1st July 2016)</h4>
<ul>
This release fixes bugs in !SUM and !SUMMARY.
<li>!SUM requests additional distributional information and the information given depends on whether the variable is a factor or a variate. But a crash occurred when a variable is declared as a factor and a variate. The fix is to treat it just as a factor (the primary definition).</li>
<li>Fix a bug causing a CRASH when the !SUMMARY qualifier is specified to request distributional information on variables and a factor (say Row) is also fitted as a variate (lin(Row)). The variable is treated as a factor</li>
</ul>
<h4>v4.1mh (10th June 2016)</h4>
<ul>
This release fixes bugs in areas including VPREDICT and GIV matrices.
<li>Better handling of correlation parameters approaching 1.0.</li>
<li>Fix calc of trace part of SCORE and of WV for DIAG term in RR.I.DIAG.</li>
<li>Honour !GF in .as over changed value in .rsv.</li>
<li>Fix failure in VPREDICT with long modelterm name.</li>
<li>Fix STACK overflow and problem forming LogDet for large GIV matrices.</li>
<li>Fix failure in VPREDICT with negative correlation in CORUH.</li>
<li>Fix failure to predict with SUBSET Trait.</li>
</ul>
<h4>v4.1mg (28th April 2016)</h4>
<ul>
The primary purpose of this release is to fix a bug, when setting level(class) names of factors that contain a decimal point.
<li>In the oats example, Nitrogen !A !L 0_cwt 0.2_cwt 0.4_cwt 0.6_cwt thus now works.</li>
<li>The special factor Trait can now be included in the PRESENT list. So PREDICT GENOTYPE !PRESENT Trait Location now works, when Location is a fixed factor nested in Trait.</li>
<li>diag(Loc).Variety now works when Loc has only 1 level.</li>
<li>An error has been corrected in a message: 'some time take' is now 'take some time'.</li>
<li>The default PIN file name has been corrected, so that it is now e.g. basename.pin rather than basename.as.pin.</li>
</ul>
<h4>v4.1mf (1st April 2016)</h4>
<ul>
This release has been modified to catch model coding errors, to fix a problem with random-number seeds, and to extend the size of the MBF structure.
<li>at(Tcode,1).ide(animal !INIT -.1) previously failed because ide was not formally recognised as a function wrapper function.</li>
<li>us(Tcode).nrm(Animal) previously failed when Tcode was a variate; a factor is expected.</li>
<li>and(a.b)us(Trait).Genotype previously failed because a SPACE (or PLUS or COMMA) is required after and(a.b). The and() model function may not interact with another term.</li>
<li>ASReml internally maintains 10 random sequences. !SEED now initializes all sequences rather than just the first.</li>
<li>The XML tag ZRatio was misspelt as Zratio in one place.</li>
<li>Previously, the number of elements in the MBF file was limited to 10,000. This limit has been removed so that a user can process a large marker file through this route. However, the GRR machinery will generally be much more efficient.</li>
<li>The format of the temporary file used when the user specified the factor labels has been modified so that up to 99,999,999 labels may be supplied (previous limit was 999,999).</li>
<li>When the pedigree was read from a data file with very long lines (more than 1000 characters), it was misread. This issue is now fixed.</li>
<li>A new variance wrapper function, rr() has been added to facilitate specification of XFA models with all zero PSIs. rr<i>k</i>(Site) is equivalent to xfa<i>k</i>(Site <i>s</i> &#42 0 &lt Initial Loadings &gt ),
where <i>s</i> is the number of levels of the factor Site.
The point of this change is that often ASReml will take much longer to fit the term xfa<i>k</i>(Site).Genotype than to fit the equivalent diag(Site).Genotype rr<i>k</i>(Site).Genotype.</li>
</ul>
<h4>v4.1mc (4th November 2015)</h4>
<ul>
The primary reasons for this update relate to problems associated with fitting large jobs. 
There have been some further changes to parameter updates, when the AI updates are not all acceptable. 
These changes reduce the iteration sequence in some jobs, especially those fitting XFA variance structures. 
They result in cosmetic differences in all jobs where the parameter update sequence is different. 
<li>A problem reading the data that occured, when data lines were very long but only a few data fields were being read, has been resolved.</li>
<li>There was a problem initializing the list of factor level labels, when more than 1 million were defined. The limit has been increased to 100 million.</li>
<li>There was a problem resulting in an ASReml crash, when plotting residuals versus fitted values from a GAMMA/INVERSE GLMM when missing values were fitted.</li>
<li>There was a problem recognising a VCM statement in the .as file, when the RESIDUAL statement was omitted.</li>
<li>The fault message, when ASReml failed to form the mixed model equations because of insufficient work space, was misleading. The messages associated with faults 333 and 315 have been revised.</li>
<li>!EMFLAG now defaults to parameter value 5 (local EM steps).</li>
<li>A job fitting a model term xfa1(Section).grm(Entry), with 8 sections and 600 Entries where grm(Entry) is dense, seemed to take longer than expected. It turned out that the equivalent model xfa1(Section 8*0 8*0.1).grm(Entry) + diag(Section).grm(Entry) ran more than 10 times faster. Setting the specific variances to zero is now equivalently done with the rr() variance function, so the equivalent model may now be written as rr1(Section).grm(Entry) + diag(Section).grm(Entry)</li>
<li>Having rectified a problem of large compute times for PEV calculation for marker effects in GBLUP analysis, the PEF values are now reported by default. Use !NOPEV on the .grr line to suppress them.</li>
</ul>
<h4>v4.1lr (10th March 2015)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.1lc - 4.1lr</li>
<li>a4data Allow MODIFY to work on nrm(Animal .01 !GU)</li>
<li>!CONTINUE and !LAST problems following qualifiers</li>
<li>Allow multiple increases in workspace allocation</li>
<li>Change to Long Integer addressing for XFA setup</li>
<li>Fix crash in VWVG with XFA large memory</li>
<li>Fix failure to handle XLT internal function in loop when large memory</li>
<li>Convert Address from short to long (W20 crash Weights)</li>
<li>Test VPREDICT has 3rd field; so no CRASH</li>
<li>Fix bug that tabulate not honoured when it follows RESIDUAL</li>
<li>Fix error; mean lines reversed in !X !Y plots</li>
<li>Add LinePrinter version to .res file</li>
<li>Stop stripping wrapper from simple terms</li>
<li>SITE*6 now becomes SITE*8</li>
<li>If name given for residual, use it e.g. units;Residual</li>
<li>Allow lit;us[1:3] in VPREDICT</li>
<li>Fix bug reading xfa(fac,2).lit</li>
<li>Fix bug parsing XFA structural</li>
<li>Insert THRLEVELS in ARGlist c.f. lg/knwd (needed for Cheese example</li>
<li>Catch bug when logical table has more than 4225000000 cells</li>
<li>Delete erroneous position message for idv(A) variance structure</li>
<li>Recognise xxx.grr Factor</li>
<li>Change to us(T).xxx;us(T) naming of components.</li>
<li>Change !CENTRE options</li>
<li>Bug recognising GenVar when GenVar created Sire*4</li>
<li>Update syntax NOKEY => NOID etc.</li>
<li>!CENTRE activated</li>
<li>Fix bug in DenDF calculation for ANTE1 in grass.as</li>
<li>Extend .xml to .pvs output</li>
<li>Extend level labels for 3-way interaction when !SLNFORM is set</li>
<li>Enable !SMODE 32</li>
<li>convert mkr/mat to .grr</li>
<li>convert snp() to grr()</li>
<li>GRR file initializes Factor class names if GIDname matches an !A variable name</li>
<li>Allow VPREDICT to use labels</li>
<li>Allow !FINAL on data line</li>
<li>Review LofDet calc of GIV</li>
<li>Fix bug reporting variance parameters in .res</li>
<li>!AIF replaces !DIAG</li>
<li>!XML to request XML output</li>
<li>Grid infill now works for multiple sections</li>
<li>Add !XML Command Line qualifier</li>
<li>Fix bug in restoring label IDV_V</li>
<li>Fix bug to eport final itern seq when simple model</li>
<li>Add !FINAL</li>
<li>Hold repeated level lists more efficiently</li>
<li>Fix bug reading in DENSE GIV of order 3189</li>
<li>Extend mvs example to !ASUV</li>
<li>Correct fix for threshold model examples (Cheese)</li>
<li>Update to .mat input option</li>
<li>Fix bug parsing !LAST Identifier.us(Trait)</li>
<li>Fix problem labelling correlations of 0/1 variates</li>
<li>Fix weighted multivariate analysis failure</li>
<li>Fix bug parsing -pSMTTWNP_30 smt (2nd P confused it)</li>
<li>Fix wrongly applied US to spl(EWEAGE) in thr 4 model</li>
<li>Fix LogL for weighted multivariate analysis</li>
<li>Fix PREDICT EWE !PRESENT BFLOCK under threshold model</li>
<li>Modify sln header layout</li>
<li>Fix bug in writing bss.vvp</li>
<li>Adjust workspace check near line 2653</li>
<li>Refine NGAMMA count in bss.as job (SING)</li>
<li>Modify SITE from A6 to A8 so better labeling in plots</li>
<li>Modif SITE from A6 to A8</li>
<li>Change bgrm to sgrm to match documentation</li>
</ul>
<h4>v4.0lb Beta (28th July 2014)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0ks - 4.0lb</li>
<li>Get start values xfa(A,1) ==> us(A)</li>
<li>Fixed bug parsing !SUBSET Tr123 Trait 1 2 3 0 0</li>
<li>Fixed bug in forming PEV of Selection Index</li>
<li>Fixed crash when invalid model term specified</li>
<li>Fixed SE reported when Singular component</li>
<li>Adjust switching from 1 to multiple factor update</li>
<li>Exclude factors from XFA eigen analysis</li>
<li>Fixed confusion parsing xfa(F,k)</li>
<li>Insert default initial values for XFA structures</li>
<li>Fixed confusion when user specifies ?sv file to read</li>
<li>Fixed bug: not opening .rsv file</li>
<li>Extended VPREDICT syntax</li>
<li>Better scaling in .vvp</li>
<li>Fixed bug parsing xfa2(Trait)</li>
<li>Fixed bug in reading named .tsv file</li>
<li>Adjustment so GDENSE works with singular GRM</li>
<li>Fixed to avoid crash when RES/PV is NaN</li>
<li>Inserted heading into .sln file</li>
<li>Fixed to avoid crash on diagnostic print when few (<40) equations</li>
<li>Adjusted format of diagnostic print</li>
<li>Fixed to print warning when $string undefined</li>
<li>Fixed new bug in !ASSIGN testing</li>
<li>Fixed new bug in testing for partial matches</li>
<li>Fix to disallow special characters in new factor names</li>
<li>Fixed format 2470, change condition</li>
<li>Prevented MKR BLUP overwriting KCV</li>
<li>Test factor labels of SUBSET better</li>
<li>Test Assign labels better</li>
<li>Increased space for GRM matrix</li>
<li>Report correct name of .rsv file used</li>
<li>Changed logical units to avoid conflict when CYCLE HOLD</li>
<li>Reduce XFAIF when MAXIT=1 (assume near solution)</li>
<li>PEV of marker blups (mkr/grm): now works</li>
<li>Extended VPREDICT to S X C D functions</li>
<li>Fixed .vvp for XFA on gamma scale</li>
<li>Fixed bug parsing labels on marker file when not many of them</li>
<li>Replaced SMIGE with SMISG so Singular GRM allowed</li>
<li>Added !PSD qualifier to declare number of singularities in GRM</li>
<li>Added !HOLD qualifier to !CYCLE so it only reads data and mkr file first cycle</li>
<li>Made !ASUV default for BINNOR model</li>
<li>Made !SPECIALCHAR parsing more robust</li>
<li>TPREDICT hyb !AVE xfa1(site) <coeffs> !ONLY xfa1(site).hyb</li>
<li>Allowed negative pivot in Dense equations (e.g. GRM)</li>
<li>Report more characteristics of GRM</li>
<li>Enabled reading labels from .mkr files</li>
<li>Improved histogram for mkr file report</li>
<li>Fixed bug when long line splits as NA*</li>
<li>Increase size of CCOL/CVAL work vectors to avoid overflow with and()</li>
<li>Fixed failure to parse and(Lhyb.xfa1(Loc),-0.2)</li>
<li>Added notice for MODIFY sat(Trial)... case</li>
<li>Revamped TSV/RSV/CONTINUE</li>
<li>Fixed bug reading .rsv when two xfa model terms</li>
<li>!FOLDER can now appear before variable definitions</li>
<li>Class names file may be in FOLDER</li>
<li>Added !SUBSECTION diagnostic warning if too many sections</li>
<li>Updated Data Sort parameters; modified report of 'size' of units</li>
<li>Fixed bug when and() term IGNORED</li>
<li>Added warning if !SUBSECTION result is suspicious</li>
<li>Added code to fix class names appearing as NA</li>
<li>Let !SECTION !ROW !COL inform record sorting if residual model does not</li>
<li>Fixed bug labelling XFA factors</li>
<li>Inserted code for if class levels NA</li>
<li>Fixed bug Recognise !A !L ../filename</li>
<li>Fixed bug in '!CONTINUE !OUTFOLDER'</li>
<li>Check CONTRAST basis variable: warning</li>
<li>Implemented new ASReml license</li>
<li>Fixed bug: failed to sort xfa.ped equations</li>
<li>Increased default space for sorting under !ROWFAC !COLFAC</li>
<li>Increased HELPFL to 200 characters</li>
<li>Fixed check of at() str too long; allow 190 rather than 90</li>
<li>Clear extra characters when expanded length > 1000</li>
<li>Changed LogDet calc message to Notice</li>
<li>In !L file, allowed "Chap 5" ==> 'Chap_5'</li>
<li>More robust LDET of sparse GIV</li>
<li>Fixed bug opening named Scratch file</li>
<li>Fixed bug when Scratch file not large enough</li>
<li>Fixed misreading long lines (character dropped)</li>
</ul>
<h4>v4.0kr Beta (26 Mar 2014)</h4>
<ul>
<li>Fixed format bug; truncated counts when NREC&lt;99999</li>
<li>Fixed bug reporting XFA matrix in .asr when some PSI zero.</li>
<li>Fixed LongInt addressing Bug reporting spl trend with Huge workspace</li>
<li>Fixed bug producing Predict PLOT with !PARALEL qualifier</li>
</ul>
<h4>v4.0kq Beta (16th March 2014)</h4>
<ul>
<li>Tighten model parsing checks and( , 0.2)</li>
<li>Allow residual units.mthr(Trait)</li>
<li>remove IF rgmstr==4 so .msv has same structure as .tsv</li>
<li>moved 'CALL StructureCheck' so ARCOL.as 18 now runs</li>
<li>modified StructureCheck to insert missing V parameter in G structures</li>
<li>Bug reporting pedigree conflict fixed, !UPPER added</li>
<li>!UPPERCASE added</li>
<li>Implement !GINDEX as TPREDICT</li>
<li>Added message when US not PD</li>
<li>Cancelled trying to reuse previous equation order: failing</li>
</ul>
<h4>v4.0kp Beta (5th March 2014)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0kn - 4.0ko</li>
<li>Fixed Bug parsing long data lines</li>
<li>SLASHSLASH restricted to .as file</li>
<li>Parent/Non-Parent labeling in aif file</li>
<li>NPD US message updated</li>
<li>GDENSE default</li>
<li>failed on at(parity).units.us(Trait)</li>
<li>Stop if more geno than markers</li>
<li>Add processing of decimal markers</li>
<li>leave log() and sqrt() wrappers</li>
<li>Fixed Bug getting DET of GIV with Groups</li>
</ul>	 
<h4>v4.0km Beta (9th January 2014)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0kk - 4.0kl</li>
<li>Fixed Bug parsing !FOR $LIST !DO command</li>
<li>Fixed Bug reading binary MBF file</li>
<li>Fix Bug creating design for row and(col)</li>
<li>!SIGMAP to allow the Sigma parameterization</li>
<li>No longer strip v from idv(units) under gamma parameterization</li>
<li>Fix Bug processing nrm(animal !INIT .25); fill in STRUC 14,16</li>
<li>Fix Bug processing idv(units.Trait) (!ASUV)</li>
<li>Added KEEP (ASUV flag) to SETSTR arg list</li>
<li>Do not create MODIFY lines for '-idv' terms</li>
<li>Set !SIGMAP for xxh() Rstructure models</li>
<li>Fix BUG in RCOV address (orange spl() error)</li>
<li>Fix fault Pedigree sort failure with Genetic Groups</li>
</ul>
<h4>v4.0kj Beta (17th January 2014)</h4>
<ul>
<li>Fix so !SUBSET Trait 3 4 5 0 0 has factor size 3</li>
<li>Fix misspelling of constraint</li>
<li>Comment out debugging statement</li> 
<li>Fix Bug resetting RSVFILE with !WORK</li>
<li>Fix Bug parsing !CONTINUE rsvfile</li>
<li>Extend !AIPENALTY</li>
<li>small changes to bsfx, mssyfa1, xfab</li> 
<li>Fix Bug switching back to all factors when 2 XFA terms!</li>
<li>Moved !DOPART $1 to top command line!</li>
<li>Increased Rstring and Selectstrg to *24</li>
<li>Inserted factor namr in title of VG plot</li>
</ul>
<h4>v4.0ki Beta (7th January 2014)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0kb - 4.0kh</li>
<li>Give heading for ar1v() term in component table</li>
<li>Fixed Bug parsing at(Age):ar1(Row):ar1(Col):us(Trait)</li>
<li>Fixed bug arising from new VC table labelling.</li>
<li>Needed to convert I4 to I8 integers KPLUSW, MEMORY </li>
<li>Long LXBASE+AIOPT in airhat call</li>
<li>Modified format 2133 and rearranged WRITEs using it</li>
<li>Fixed bug reporting conflicting pedigree (labelling)</li>
<li>Fixed bug parsing <TAB><TAB> when reading pedigree</li>
<li>Modified variance components table at request of Robin extensive output changes.</li>
<li>Bug: inserted MOD() for case of -ve mdel term</li>
<li>Extended functionality of VCM for Christ Brien</li>
<li>Extended VCM statement to allow list of parameters in place [VCB]</li>
<li>reordered so license check before prompts</li>
<li>Do not set space for data sorting if not required</li>
<li>Do not zero sln(EQN,2) as it sometimes crashes asreml-r</li>
<li>Change base 16000 to 17500 so can have more than 1000 loadings</li>
<li>Allow I9 for counts if MROW > 99999 (else I6)</li>
<li>!PRECISION 1 2 3 === 6 5 4</li>
<li>Assume random for nrm, grm and vs structures</li>
<li>!SAVE equivalent/alternative to !GIV</li>
<li>!L filename !SKIP equivalent to !LSKIP</li> 
<li>Accommodate spurious leading BLANKS</li>
<li>Allow blank lines in files being merged</li>
<li>Notify if no Trait with !MULTINOMIAL</li>
<li>Changed comment</li>
<li>Changed LU=8 to LU=13 so TABULATE works!</li>
<li>Revampted Variance component table.</li>
<li>Bug adjusting arh(row)</li>
<li>Adjusted MP absorption so return correct Fincr</li>
<li>Fixed bug parsing ped col labels ANIMAL,SIRE,DAM</li>
<li>Improved error messages</li>
<li>Inserted links for !SUBSECTION in new syntax</li>
<li>Avoided bad memory reference</li>
<li>Fix format bug</li>
<li>Allow '*' on next line parsing old format Structure lines </li>
<li>Bug parsing !CONT 3</li>
<li>Bug reporting restart from .msv</li>
<li>Bug parsing str(row</li>
<li>Added RSVFILE</li>
<li>Allowed filename arg to !CONTINUE for .rsv file</li>
<li>Added warnings in two places where // notation fails.</li>
<li>Fix labelling of Matern parameters - Matern.as</li>
</ul>

<h4>v4.0ka Beta (08th October 2013)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0jy - 4.0jz</li>
<li>Extended code to permit residual diag().ar1().ar1()</li>
<li>Inserted 'Notice:' in diagnostic message</li>
<li>Fixed bug in Peppers job by moving VSOLN(EQN)=XD</li>
<li>Modified labelling of pedigree counts table.</li>
<li>Fixed bug wrt flevel(quadGF) when quadGF is zero</li>
<li>Treat MKR data outside -2,2 as missing.</li>
<li>Faster parsing of marker file values 0,1,2</li>
<li>Fixed bug in default skipping of first field of .mkr file</li>
<li>Fixed bug making file names for !out option with !RENAME cycling</li>
<li>Fixed bug when Insert Exit path when !A !L labels file not present.</li>
<li>Read error diagnostic print added</li>
<li>Added links for !EXCLUDE and !KCV</li>
<li>Added actions for !EXCLUDE </li>
<li>Added reading !EXCLUDE and !KCV qualifiers</li>
<li>Added KCV functionality</li>
<li>Added KCV and ExCLUDE variables</li>
<li>Fixed bug due to unset variable KS.</li>
</ul>

<h4>v4.0jx Beta (08th September 2013)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0ju - 4.0jw</li>
<li>Fixed bug in LogL calc for NSECT>1 when NSPAT>1<p>[Modifies reported LogL in dhglm, az06, binnor] [Forum query re spatial LogL]</p></li>
<li>Fixed bug with string overflow when parsing CALC statements (ININTR)</li>
<li>Get GLMM weights in correct order when data records sorted</li>
<li>Suppress erroneous messages parsing pedigree with !MGS option</li>
<li>Take PARENT Labels from 1st line of ped file where provided.</li>
<li>Move .msv file to logical unit 18.</li>
<li>Adjust XFA update cycling parameters.</li>
<li>Fixed bug when GLMM and !{ !} fixed.</li>
<li>Revise XFAIF to incorporate !AIPENALTY</li> 
<li>Adjusted limits for pi in !FBB</li>
<li>Tidy up some labeling of VC table<p>This affects most .asr files as the 'Variance' label has been changed to 'Residual'</p></li>
<li>Adjust model label strings</li>
<li>!MAXIT -1 aborts run after parsing job and writing .tsv</li>
<li>Added !TSV and !MSV as alternatives to !CONTINUE 2/3</li>
<li>RPscale defaults to 1.0 if supplied as zero</li>
<li>Supply better start values for US residual in !ASUV case</li>
<li>Fixed bug in processing singular VCM matrix</li>
<li>Changed reporting and reading of unset RP fields</li>
<li>Changed so !MAXIT 0 just parses the .as file, writes .tsv</li>
<li>Added !FBB pi to explore Bayes B like models</li>
<li>Changed output heading to Variance Component table affects all jobs</li>
<li>Changed output labeling of components affects SpatialModel jobs</li>
<li>Modified XFA update rules changes Factor Analytic jobs</li>
<li>Modified contents of .tsv affects all jobs</li>
<li>New .msv file for all jobs</li>
</ul>

<h4>v4.0jt Beta (8th July 2013)</h4>
<ul>
<li>fixed license problem in ASReml-W</li>
</ul>

<h4>v4.0jt Beta (2nd July 2013)</h4>
<ul>
<li>fixed bug reporting Boundary parameter as S</li>
<li>fixed bug reading !SPATIAL</li>
<li>fixed bug failing to properly initialize some parameters</li>
<li>fixed Test for label 1E-8</li>
<li>fixed TXTFORM wrong for multiple tabulates</li>
<li>fixed correlation close to boundary</li>
<li>fixed bug plotting residuals from AR.AR.US model</li>
<li>fixed bug parsing residual AR1(R).ar1(C).us(Tr)</li>
<li>fixed bug making predict design for dev(Year).us(Trait)</li>
</ul>

<h4>v4.0js Beta (14th June 2013)</h4>
<ul>
<li>Cumulative enhancements from internal version 4.0jf - 4.0jr</li>
<li>Fixed problem when bivariate weighted used !TOT !WT</li>
<li>Fixed problem of at(Trait,2).X anulling Y for other traits when X missing</li>
<li>Added warning when CSV file has less fields than expected</li>
<li>Allowed qualifiers on idv() terms</li>
<li>Allow !PART list to extend beyond character 80</li>
<li>Honour !VARSCALE for 'residual units' and allow on model line</li>
<li>Fix bug parsing at(evt,'SFM_138)</li>
<li>Fix bug fixing ORDER parsing at(evt,'SFM_138)</li>
<li>Extend !CONTINUE 2 to read .tsv file</li>
<li>Get better default initial values for XFA models</li>
<li>Fix bug in format of MODIFY error message</li>
<li>Fix bug putting AINV structure on geno in loc.geno</li>
<li>Fix bug sorting data for spatial residual multi sections</li>
<li>!PRECISION ; adjust singularity detection threshold in GRM inversion</li>
<li>Change default init var under variance scale</li>
<li>Added !GDENSE qualifier to process GRM() densly</li>
<li>Add !OUTFOLDER qualifier to top command line</li>
<li>Fix bugs processing cholkc() wrapper function</li>
<li>Fixed bugs parsing spherical and circular models</li>
<li>Changed scatter plot to grid base when > 10000 points</li>
<li>Added Mean Axes to XYG scatter plot</li>
<li>Honour !GU after at(G,1).parc</li>
<li>Fix bug in reading G structure terms from new format .rsv file.</li>
<li>Add message when explicit V in R structure removed</li>
<li>Add !CENTRE qualifier, revise weight expression</li>
</ul>

<h4>v4.0je Beta (29th May 2013)</h4>
<ul>
<li>Fixed some license problems.</li>
</ul>

<h4>v4.0jd Beta (4th March 2013)</h4>
<ul>
<li>Beta release of ASReml 4.</li>
</ul>

</td><td width="20"></td></tr>
<tr><td/><td/><td/><td/><td/><td><br><br></td></tr>
</table>

<span class="footer">&nbsp;&nbsp;&copy; Copyright 2010-2015 - VSN International Ltd.</span>
</table>

</body>
</html>